josh-proxy/src/bin/josh-proxy.rs

```diff
@@ -397,13 +398,32 @@ async fn call_service(
     let parsed_url = {
         if let Some(parsed_url) = FilteredRepoUrl::from_str(&path) {
             let mut pu = parsed_url;
+
+            if pu.rest.starts_with(":") {
+                let guessed_filter = pu.rest.trim_end_matches("/info/refs");
+                return Ok(error_response(
+                    hyper::Body::from(formatdoc!(
+                        r#"
+                        Invalid filter: "{0}"
+
+                        Note: filter definitions should end with ".git":
```

Comment:
Note quite accurate: `.git` is not part of the filter definition, but of the url format.
So the message should be something like:
`Invalid URL: {0}`
`Note: Repository URLs should end with ".git"`